extension/sqlsmith/statement_generator.cpp

```diff
@@ -109,6 +150,76 @@ unique_ptr<CreateStatement> StatementGenerator::GenerateCreate() {
 	return create;
 }
 
+unique_ptr<AttachStatement> StatementGenerator::GenerateAttach() {
+	auto attach = make_uniq<AttachStatement>();
+	attach->info = GenerateAttachInfo();
+	return attach;
+}
+
+unique_ptr<DetachStatement> StatementGenerator::GenerateDetach() {
+	auto detach = make_uniq<DetachStatement>();
+	detach->info = GenerateDetachInfo();
+	return detach;
+}
+
+// generate USE statement
+unique_ptr<SetStatement> StatementGenerator::GenerateSet() {
+	auto name_expr = make_uniq<ConstantExpression>(Value("RANDOM_STRING_" + RandomString(10)));
+	if (RandomPercentage(90)) {
+		auto name = RandomNameFromContext();
+		name_expr = make_uniq<ConstantExpression>(Value(name));
+	}
+	auto set = make_uniq<SetVariableStatement>("schema", std::move(name_expr), SetScope::AUTOMATIC);
+	return set;
+}
+
+unique_ptr<MultiStatement> StatementGenerator::GenerateAttachUse() {
+	auto multi_statement = make_uniq<MultiStatement>();
+	multi_statement->statements.push_back(std::move(GenerateAttach()));
+	multi_statement->statements.push_back(std::move(GenerateSet()));
+	return multi_statement;
+}
+
+//===--------------------------------------------------------------------===//
+// Generate Detach Info
+//===--------------------------------------------------------------------===//
+
+unique_ptr<DetachInfo> StatementGenerator::GenerateDetachInfo() {
+	auto info = make_uniq<DetachInfo>();
+	if (RandomPercentage(20)) {
+		info->name = "RANDOM_NAME_" + RandomString(15);
+	} else {
+		info->name = RandomNameFromContext();
+	}
+	return info;
+}
+
+std::string StatementGenerator::RandomNameFromContext() {
+	auto state = GetDatabaseState(context);
```

Comment:
I would give this function a different name. Maybe `GetRandomAttachedDataBase()`?